confgenerator/prometheus.go

```diff
@@ -221,7 +218,7 @@ func checkTLSConfig(tlsConfig commonconfig.TLSConfig) error {
 
 // validatePrometheus checks the receiver configuration is valid.
 func validatePrometheus(promConfig promconfig.Config) (string, error) {
-	if PrometheusFeatureGate != "enabled" {
+	if !IsExperimentalReceiverEnabled("prometheus") {
```

Comment:
Can we define the list of featuregates (like `prometheus`) somewhere central? Perhaps a separate file?

Reply:
I'm not sure what the goal of a predefined list would be -- are you looking for validating the list in case the user supplies an invalid feature gate?

Reply:
I was thinking the purpose was twofold:

- If we have a file that lists all the feature flags we support, we have a defined naming convention of what the feature flags should look like and have one place to find them all.
- We could validate that user supplies a valid feature flag that the agent supports.

Reply:
I tried this and I ended up with quite a bit of additional complexity. IMO it's not worth doing this considering experimental features are unsupported, not user-facing, and transient in nature -- we'll be GA-ing both prometheus and OTLP in the coming months, which means all of this can be reverted at that point anyway.

Nonetheless, I pushed it to [`jefferbrecht-refactor-receiver-gate-alternate`](https://github.com/GoogleCloudPlatform/ops-agent/compare/jefferbrecht-refactor-receiver-gate...jefferbrecht-refactor-receiver-gate-alternate) if you'd like to take a look. LMK if you see any opportunities to simplify it.

Reply:
I think your explanation is pretty sound. lets ship this as is and revisit this if the needs grow in the future. Normally we go to GA directly, so I imagine this is not a problem for a while.